[query] automatic retry code in Java is broken

### What happened?

https://batch.hail.is/batches/8043502/jobs/43724

```
Caused by: java.lang.IllegalArgumentException: bound must be positive
	at java.util.Random.nextInt(Random.java:388)
	at scala.util.Random.nextInt(Random.scala:70)
	at is.hail.services.package$.delayMsForTry(package.scala:47)
	at is.hail.services.package$.retryTransientErrors(package.scala:186)
	at is.hail.io.fs.GoogleStorageFS$$anon$1.retryingRead(GoogleStorageFS.scala:220)
	at is.hail.io.fs.GoogleStorageFS$$anon$1.readHandlingRequesterPays(GoogleStorageFS.scala:226)
	at is.hail.io.fs.GoogleStorageFS$$anon$1.fill(GoogleStorageFS.scala:257)
	at is.hail.io.fs.FSSeekableInputStream.read(FS.scala:170)
	at java.io.DataInputStream.read(DataInputStream.java:149)
	at is.hail.utils.ByteTrackingInputStream.read(ByteTrackingInputStream.scala:28)
	at is.hail.utils.richUtils.RichInputStream$.readRepeatedly$extension0(RichInputStream.scala:21)
	at is.hail.utils.richUtils.RichInputStream$.readFully$extension1(RichInputStream.scala:12)
	at is.hail.io.StreamBlockInputBuffer.readBlock(InputBuffers.scala:549)
	at is.hail.io.ZstdInputBlockBuffer.readBlock(InputBuffers.scala:643)
	at is.hail.io.BlockingInputBuffer.ensure(InputBuffers.scala:384)
	at is.hail.io.BlockingInputBuffer.readByte(InputBuffers.scala:402)
	at is.hail.io.LEB128InputBuffer.readByte(InputBuffers.scala:219)
	at __C372collect_distributed_array_matrix_native_writer.__m478readLeafNode(Unknown Source)
	at __C372collect_distributed_array_matrix_native_writer.apply_region16_290(Unknown Source)
	at __C372collect_distributed_array_matrix_native_writer.apply_region4_318(Unknown Source)
	at __C372collect_distributed_array_matrix_native_writer.apply_region2_501(Unknown Source)
	at __C372collect_distributed_array_matrix_native_writer.apply(Unknown Source)
	at __C372collect_distributed_array_matrix_native_writer.apply(Unknown Source)
	at is.hail.backend.BackendUtils.$anonfun$collectDArray$16(BackendUtils.scala:91)
	at is.hail.utils.package$.using(package.scala:637)
	at is.hail.annotations.RegionPool.scopedRegion(RegionPool.scala:162)
	at is.hail.backend.BackendUtils.$anonfun$collectDArray$15(BackendUtils.scala:90)
	at is.hail.backend.service.Worker$.$anonfun$main$12(Worker.scala:167)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at is.hail.services.package$.retryTransientErrors(package.scala:182)
	at is.hail.backend.service.Worker$.$anonfun$main$11(Worker.scala:166)
	at is.hail.backend.service.Worker$.$anonfun$main$11$adapted(Worker.scala:164)
	at is.hail.utils.package$.using(package.scala:637)
	at is.hail.backend.service.Worker$.main(Worker.scala:164)
	at is.hail.backend.service.Main$.main(Main.scala:14)
	at is.hail.backend.service.Main.main(Main.scala)
	... 11 more

```

### Version

0.2.124

### Relevant log output

_No response_

--- a/hail/src/main/scala/is/hail/services/package.scala
+++ b/hail/src/main/scala/is/hail/services/package.scala
@@ -42,10 +42,10 @@ package object services {
     // Based on AWS' recommendations:
     // - https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/
     // - https://github.com/aws/aws-sdk-java/blob/master/aws-java-sdk-core/src/main/java/com/amazonaws/retry/PredefinedBackoffStrategies.java
-    val multiplier = 1 << math.min(tries, LOG_2_MAX_MULTIPLIER)
-    val ceilingForDelayMs = baseDelayMs * multiplier
+    val multiplier = 1L << math.min(tries, LOG_2_MAX_MULTIPLIER)
+    val ceilingForDelayMs = math.min(baseDelayMs * multiplier, maxDelayMs).toInt
     val proposedDelayMs = ceilingForDelayMs / 2 + Random.nextInt(ceilingForDelayMs / 2 + 1)
-    return math.min(proposedDelayMs, maxDelayMs)
+    return proposedDelayMs
   }
 
   def sleepBeforTry(

